oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/celliterator/CellIterator.java

```diff
@@ -526,74 +545,106 @@ public IterateAllEntry(
    *         intervals.
    */
   public Stream<IterateAllEntry> iterateByContribution(GridOSHEntity cell) {
-    OSHDBTimestampInterval timeInterval = new OSHDBTimestampInterval(timestamps);
-
+    var cellBoundingBox = XYGrid.getBoundingBox(new CellId(
+        cell.getLevel(),
+        cell.getId()
+    ), true);
     final boolean allFullyInside;
     if (isBoundByPolygon) {
-      // if cell is fully inside bounding box/polygon we can skip all entity-based inclusion checks
-      OSHDBBoundingBox cellBoundingBox = XYGrid.getBoundingBox(new CellId(
-          cell.getLevel(),
-          cell.getId()
-      ), true);
       if (bboxOutsidePolygon.test(cellBoundingBox)) {
         return Stream.empty();
       }
       allFullyInside = bboxInPolygon.test(cellBoundingBox);
     } else {
       allFullyInside = false;
     }
+    return iterateByContribution(cell.getEntities(), allFullyInside);
+  }
+
+  /**
+   * Helper method to easily iterate over all entity modifications in a cell that match a given
+   * condition/filter.
+   *
+   * @param cellData the entities to iterate through
+   * @param allFullyInside
+   *
+   * @return a stream of matching filtered OSMEntities with their clipped Geometries and timestamp
+   *         intervals.
+   */
+  public Stream<IterateAllEntry> iterateByContribution(Iterable<? extends OSHEntity> cellData,
+      boolean allFullyInside) {
+    OSHDBTimestampInterval timeInterval = new OSHDBTimestampInterval(timestamps);
 
     if (includeOldStyleMultipolygons) {
       //todo: remove this by finishing the functionality below
       throw new UnsupportedOperationException("this is not yet properly implemented (probably)");
     }
+    return Streams.stream(cellData)
+        .filter(oshEntity -> allFullyInside || (oshEntity.intersects(boundingBox)
+            && !(isBoundByPolygon && bboxOutsidePolygon.test(oshEntity))))
+        .filter(oshEntityPreFilter)
```

Comment:
In case of a `isBoundByPolygon` this is right.
I will spilt this (ugly ;-)) bbox filter in actually two filters
1. bbox test
2. entity test
3. bboxOutsidePolygon test

Reply:
done filter is now this:
```java
   .filter(oshEntity -> allFullyInside || oshEntity.intersects(boundingBox))
   .filter(oshEntityPreFilter)
   .filter(oshEntity -> allFullyInside || !isBoundByPolygon || !bboxOutsidePolygon.test(oshEntity))
 ```